src/routes/pois/containers/PoisPage.js

```diff
@@ -37,9 +37,7 @@ export class PoisPage extends React.Component<PropsType> {
   renderPoiListItem = (poi: PoiModel) => <PoiListItem key={poi.path} poi={poi} />
 
   redirectToPath = (path: string) => {
-    const action = pathToAction(path, this.props.routesMap)
-    setKind(action, 'push')
-    this.props.dispatch(action)
+    this.props.redirect(pathToAction(path, this.props.routesMap))
```

Comment:
this does not push the url on top of the stack, but replace the current entry with `path`.
(So if you click on a link and navigate back, the browser will navigate two pages back)